service/src/main/java/bio/terra/workspace/service/petserviceaccount/PetSaService.java

```diff
@@ -147,42 +155,49 @@ public Policy enablePetServiceAccountImpersonation(UUID workspaceId, UserWithPet
    *
    * @param workspaceId ID of the workspace to disable pet SA in
-   * @param userAndPet The user and pet pair losing access to the pet SA
+   * @param userToDisableEmail The user losing access to pet SA
+   * @param userRequest This request's token will be used to authenticate SAM requests
    * @param eTag GCP eTag which must match the pet SA's current policy. If null, this is ignored.
    */
   public Optional<Policy> disablePetServiceAccountImpersonationWithEtag(
-      UUID workspaceId, UserWithPetSa userAndPet, @Nullable String eTag) {
+      UUID workspaceId,
+      String userToDisableEmail,
+      AuthenticatedUserRequest userRequest,
+      @Nullable String eTag) {
+    String proxyGroupEmail =
+        SamRethrow.onInterrupted(
+            () -> samService.getProxyGroupEmail(userToDisableEmail, userRequest.getRequiredToken()),
+            "disablePet");
 
     String projectId = gcpCloudContextService.getRequiredGcpProject(workspaceId);
-    ServiceAccountName petServiceAccount =
-        ServiceAccountName.builder().email(userAndPet.getPetEmail()).projectId(projectId).build();
     try {
+      Optional<ServiceAccountName> userToDisablePetSA =
+          getUserPetSa(projectId, userToDisableEmail, userRequest);
+      if (userToDisablePetSA.isEmpty()) {
+        return Optional.empty();
+      }
+
       Policy saPolicy =
           crlService
               .getIamCow()
               .projects()
               .serviceAccounts()
-              .getIamPolicy(petServiceAccount)
+              .getIamPolicy(userToDisablePetSA.get())
               .execute();
       // If the caller supplied a non-null eTag value, it must match the current policy in order to
       // modify the policy here. Otherwise, return without modifying the policy to prevent
       // clobbering other changes.
       if (eTag != null && !saPolicy.getEtag().equals(eTag)) {
         logger.warn(
             "GCP IAM policy eTag did not match expected value when revoking pet SA access for user {} in workspace {}. This is normal for Step retries.",
-            userAndPet.getUserEmail(),
+            userToDisableEmail,
             workspaceId);
         return Optional.empty();
       }
-      // TODO(PF-991): when enablePetServiceAccountImpersonation stops putting the pet SA in this
-      //  binding, this method should stop removing it.
       Binding bindingToRemove =
           new Binding()
               .setRole(SERVICE_ACCOUNT_USER_ROLE)
-              .setMembers(
-                  ImmutableList.of(
-                      "user:" + userAndPet.getUserEmail(),
-                      "serviceAccount:" + petServiceAccount.email()));
+              .setMembers(ImmutableList.of("group:" + proxyGroupEmail));
       // If no bindings exist, getBindings() returns null instead of an empty list. If there are
       // no policies, there is nothing to revoke, so this method is finished.
       List<Binding> oldBindingList = saPolicy.getBindings();
```

Comment:
Aside: it might be nice to have a wrapper method like `getBindingsList()` that returned an empty list instead of null.